pkg/policy/groups/aws/aws.go

```diff
@@ -0,0 +1,135 @@
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/cilium/cilium/pkg/policy/api"
+)
+
+const (
+	awsLogLevel         = aws.LogOff // For debugging pourposes can be set to aws.LogDebugWithSigning
+	awsDefaultRegionKey = "AWS_DEFAULT_REGION"
+	awsDefaultRegion    = "eu-west-1"
+)
+
+var (
+	policySecurityGroupIdKey = aws.String("instance.group-id")
+	policySecurityGroupName  = aws.String("instance.group-name")
+	policyEC2Labelskey       = "tag"
+)
+
+func init() {
+	api.RegisterToGroupsProvider(api.AWSProvider, GetIPsFromGroup)
+}
+
+// GetIpsFromGroup will return the list of the ips for the given group filter
+func GetIPsFromGroup(group *api.ToGroups) ([]net.IP, error) {
+	result := []net.IP{}
+	if group.Aws == nil {
+		return result, fmt.Errorf("no aws data available")
+	}
+	ips, err := getInstancesIpsFromFilter(group.Aws)
+	return ips, err
+}
+
+// initializeAWSAccount retrieve the env variables from the runtime and it
+// iniliazes the account in the specified region.
+func initializeAWSAccount(region string) (*aws.Config, error) {
+	cfg, err := external.LoadDefaultAWSConfig()
+	if err != nil {
+		return nil, fmt.Errorf("Cannot initialize aws connector: %s", err)
+	}
+	cfg.Region = region
+	cfg.LogLevel = awsLogLevel
+	return &cfg, nil
+}
+
+// getInstancesFromFilter returns the instances IPs in aws EC2 filter by the
+// given filter
+func getInstancesIpsFromFilter(filter *api.AWSGroups) ([]net.IP, error) {
+	region := filter.Region
+	if filter.Region == "" {
+		region = getDefaultRegion()
+	}
+	input := &ec2.DescribeInstancesInput{}
+	for labelKey, labelValue := range filter.Labels {
+		newFilter := ec2.Filter{
+			Name:   aws.String(fmt.Sprintf("%s:%s", policyEC2Labelskey, labelKey)),
+			Values: []string{labelValue},
+		}
+		input.Filters = append(input.Filters, newFilter)
+	}
+	if len(filter.SecurityGroupsIds) > 0 {
+		newFilter := ec2.Filter{
+			Name:   policySecurityGroupIdKey,
+			Values: filter.SecurityGroupsIds,
+		}
+		input.Filters = append(input.Filters, newFilter)
+	}
+	if len(filter.SecurityGroupsNames) > 0 {
+		newFilter := ec2.Filter{
+			Name:   policySecurityGroupName,
+			Values: filter.SecurityGroupsNames,
+		}
+		input.Filters = append(input.Filters, newFilter)
+	}
+	cfg, err := initializeAWSAccount(region)
+	if err != nil {
+		return []net.IP{}, err
+	}
+	svc := ec2.New(*cfg)
+	req := svc.DescribeInstancesRequest(input)
+	result, err := req.Send()
+	if err != nil {
+		return []net.IP{}, fmt.Errorf("Cannot retrieve aws information: %s", err)
+	}
+	return awsDumpIpsFromRequest(result), nil
+}
+
+// getDefaultRegion returns the given region of the default one.
+// @TODO retrieve the region from aws metadata.
+func getDefaultRegion() string {
+	val := os.Getenv(awsDefaultRegionKey)
+	if val != "" {
+		return val
+	}
+	return awsDefaultRegion
+}
+
+func awsDumpIpsFromRequest(req *ec2.DescribeInstancesOutput) []net.IP {
+	result := []net.IP{}
+	if len(req.Reservations) == 0 {
```

Comment:
you can remove these 3 lines, the for loop on line 122 is not executed if `len(req.Reservations) == 0`